internal/events/kafka.go

```diff
@@ -4,30 +4,94 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"strings"
+
+	"github.com/segmentio/kafka-go/sasl/plain"
 
 	"github.com/segmentio/kafka-go"
 	"github.com/stellar/go/support/log"
 	"golang.org/x/exp/maps"
 )
 
+type KafkaSecurityProtocol string
+
+const (
+	KafkaProtocolPlaintext     KafkaSecurityProtocol = "PLAINTEXT"
+	KafkaProtocolSASLPlaintext KafkaSecurityProtocol = "SASL_PLAINTEXT"
+	KafkaProtocolSASLSSL       KafkaSecurityProtocol = "SASL_SSL"
+	KafkaProtocolSSL           KafkaSecurityProtocol = "SSL"
+)
+
+func ParseKafkaSecurityProtocol(protocol string) (KafkaSecurityProtocol, error) {
+	protocol = strings.ToUpper(protocol)
+	switch KafkaSecurityProtocol(protocol) {
+	case KafkaProtocolPlaintext, KafkaProtocolSASLPlaintext, KafkaProtocolSASLSSL, KafkaProtocolSSL:
+		return KafkaSecurityProtocol(protocol), nil
+	default:
+		return "", fmt.Errorf("invalid kafka security protocol: %s", protocol)
+	}
+}
+
+type KafkaConfig struct {
+	Brokers          []string
+	SecurityProtocol KafkaSecurityProtocol
+	SASLUsername     string
+	SASLPassword     string
+}
+
+func (kc *KafkaConfig) Validate() error {
+	if len(kc.Brokers) == 0 {
+		return fmt.Errorf("brokers cannot be empty")
+	}
+
+	if kc.SecurityProtocol == "" {
+		return fmt.Errorf("security protocol cannot be empty")
+	}
+
+	if kc.SecurityProtocol == KafkaProtocolSASLPlaintext || kc.SecurityProtocol == KafkaProtocolSASLSSL {
```

Comment:
FYI: you can also use slices.Contains for this kind of comparison, like we do in:
https://github.com/stellar/stellar-disbursement-platform-backend/blob/a50459672d50947e154aeae36b16c17fc2e34578/internal/transactionsubmission/utils/errors.go#L304-L306

Reply:
Addressed here [4670680](https://github.com/stellar/stellar-disbursement-platform-backend/pull/162/commits/467068017a2e0586ee11f441c69ab802189cc5fa)